### PR TITLE
[JBPM-8501] Allow reloading KIE Server clients

### PR DESCRIPTION
**Pull request by ruromero**
Fix https://issues.jboss.org/browse/JBPM-8501

UI changes:
* Allow refresh the KieServer lists and disable those whose state is not "SUCCESS"
* Run migrations radio button is by default "Now"
* Gradually moving from axios to fetch and restful calls
* Notify the users when the process definitions cannot be retrieved

Backend changes:
* Async retry to create the KieServer clients and reload status each time the kieServers status is requested
* Pending reformat in some files
* ProcessDefinitions fetching are now per process

Signed-off-by: Ruben Romero Montes <rromerom@redhat.com>

### CODE REVIEW

**Comment by jstastny-cz** [comment]
@ruromero  found out following issue in the PR:
1. create a migration plan when kie-server runs
2. stop kie-server
3. click execute near the migration plan -> nothing happens - no indication that sth went wrong, no logs about reconnect attempt in console
4. start kie-server again, the execute button starts working.

Could you add a notification to the execute action?

**Comment by ruromero** [comment]
@jstastny-cz done

File: process-migration-service/frontend/src/component/tabMigrationPlan/MigrationPlansBase.js [hunk 1]
```diff
@@ -56,6 +63,11 @@ export default class MigrationPlansBase extends React.Component {
       .then(function() {
         self.retrieveAllPlans();
         self.hideDeleteDialog();
+      })
+      .catch(() => {
```
Comment (jstastny-cz) [inline process-migration-service/frontend/src/component/tabMigrationPlan/MigrationPlansBase.js]: There is no invocation of kie-server in this function, right? The catch block provides wrong explanation, might be redundant.

File: process-migration-service/frontend/src/component/tabMigrationPlan/MigrationPlansBase.js [hunk 2]
```diff
@@ -24,10 +24,17 @@ export default class MigrationPlansBase extends React.Component {
...
-      const plans = res.data;
-      this.setState({ plans, filteredPlans: plans });
-    });
+    axios
+      .get(servicesUrl, {})
+      .then(res => {
+        const plans = res.data;
+        this.setState({ plans, filteredPlans: plans });
+      })
+      .catch(() => {
```
Comment (jstastny-cz) [inline process-migration-service/frontend/src/component/tabMigrationPlan/MigrationPlansBase.js]: there's no communication with kieserver in this function, the catch block does not apply to kieserver failure.

File: process-migration-service/frontend/src/component/tabMigrationPlan/MigrationPlansBase.js [hunk 3]
```diff
@@ -84,6 +96,11 @@ export default class MigrationPlansBase extends React.Component {
           addPlanResponseJsonStr: JSON.stringify(response.data, null, 2)
         });
         self.retrieveAllPlans();
+      })
+      .catch(() => {
```
Comment (jstastny-cz) [inline process-migration-service/frontend/src/component/tabMigrationPlan/MigrationPlansBase.js]: There should be no invocation of kieserver in the function.

File: process-migration-service/frontend/src/component/tabMigrationPlan/wizardAddPlan/PageDefinition.js [hunk 4]
```diff
@@ -1,41 +1,90 @@
...
+        callbackFn(res);
+      })
+      .catch(err => {
+        const status = err.status;
+        let errorMsg;
+        if (status === 404) {
+          if (containerId !== "" || processId !== "") {
+            errorMsg = `Process not found: ${containerId}/${processId}`;
+          } else {
+            errorMsg = "Provide a valid containerId and processId";
```
Comment (jstastny-cz) [inline process-migration-service/frontend/src/component/tabMigrationPlan/wizardAddPlan/PageDefinition.js]: We should check the values before invoking the endpoint.

File: process-migration-service/frontend/src/component/tabMigrationPlan/wizardAddPlan/PageDefinition.js [hunk 5]
```diff
@@ -1,41 +1,90 @@
...
-          sourceProcessId: this.props.sourceProcessId,
-          sourceContainerId: this.props.sourceContainerId,
-          targetProcessId: this.props.targetProcessId,
-          targetContainerId: this.props.targetContainerId,
-          kieServerId: this.props.kieServerIds
+  getDefinition = (containerId, processId, callbackFn) => {
+    const defURL = `${BACKEND_URL}/kieserver/${
+      this.props.kieServerId
+    }/definitions/${containerId}/${processId}`;
+    fetch(defURL, {
```
Comment (jstastny-cz) [inline process-migration-service/frontend/src/component/tabMigrationPlan/wizardAddPlan/PageDefinition.js]: again using fetch vs. axios.get() ... if there's not any reason to mix the two, I'd stick with one way.

File: process-migration-service/frontend/src/component/tabMigrationPlan/MigrationPlansBase.js [hunk 6]
```diff
@@ -102,6 +119,11 @@ export default class MigrationPlansBase extends React.Component {
...
```
Comment (jstastny-cz) [inline process-migration-service/frontend/src/component/tabMigrationPlan/MigrationPlansBase.js]: is there any invocation of a kie-server?

**Comment by ruromero** [comment]
@MarianMacik @jstastny-cz rebase done

**Comment by ruromero** [comment]
jenkins retest this

**Pull request merged**